### ISSUE TITLE
Common CSS between patterns are getting duplicated

### ISSUE THREAD

**Issue by kexuibm** [carbon-design-system/ibm-dotcom-library#1743]
Using V1.5.0, we are currently generating a css file per IDL Pattern (Content Block or Section).  This CSS file is generated using the IDL's pattern's required SCSS file.  In Production, these css files would be aggregated into a single file and loaded by the rendered page. 

However it looks like each of these generated CSS files has lots of underlying shared styles with each other.  So in taking the current approach, that aggregated css file would be much larger than it has to be.  This may affect page performance in Production.  So the ask is for DDS to find/provide a way to separate the common css styles needed by all IDL patterns, from the pattern specific css styles.

**Comment by ljcarot** [comment]
@kexuibm Changing this to a question.  This may have implications for other adopters of patterns beyond Drupal.  We'll discuss with team.  Have you seen page performance issues already?

**Comment by jeffchew** [comment]
@kexuibm I'm also wondering if this is something happening with the artifact creation setup in webpack rather than what's in the library. Typically, webpack handles the code duplication when creating one or many artifacts, and ensures that all the code necessary has everything it needs, whether if it's a single artifact, multiple artifacts, dynamic imports, etc. We might have to review what your webpack configuration setup is and take it from there, but from the IBM.com Library side, the scss imports are all referencing the necessary dependencies discretely in each of their respective pattern/component level scss.

cc: @kennylam @annawen1 @ljcarot @photodow @RaphaelAmadeu

**Comment by kexuibm** [comment]
> @kexuibm I'm also wondering if this is something happening with the artifact creation setup in webpack rather than what's in the library. Typically, webpack handles the code duplication when creating one or many artifacts, and ensures that all the code necessary has everything it needs, whether if it's a single artifact, multiple artifacts, dynamic imports, etc. We might have to review what your webpack configuration setup is and take it from there, but from the IBM.com Library side, the scss imports are all referencing the necessary dependencies discretely in each of their respective pattern/component level scss.
> 
> cc: @kennylam @annawen1 @ljcarot @photodow @RaphaelAmadeu

@jeffchew  Yes, we've been using the scss imports per pattern/subpattern from the story book readme documentation for each pattern.  I think the issue is that we are generating separate css files one for each pattern.  One idea that we've been investigating is to simply create a webpack configuration that combines all of the scss imports from all the patterns in learn page to see if that removes duplicates classes.
